docs/contributor/arch/019-switch-from-gateways-to-a-central-agent.md

```diff
@@ -0,0 +1,107 @@
+
+  Pushing OTLP data from an application to the gateway running with replicas requires load balancing. Unfortunately, a Kubernetes service is acting on L4 while GRPC/HTTP2 is on L7, so an additional load balancer is needed. That can be easily solved by using Istio, which becomes a requirement for applications. However, some applications don't use Istio (StatefulSets often don't) but want to provide OTLP data.
+
+- Istio is mandatory for mTLS
+
+  App to Gateway communication is across nodes and with that should use mTLS, so Istio needs to be used which sometimes is not possible. Also, Istio causes additional overhead like noise filtering and collecting observability data about the service mesh should not rely on the service mesh.
+
+- Cumulativetodelta transformation not supported
+
+  This transformation is mandatory for Dynatrace support and requires that data from a Pod gets processed always by the same instance.
+
+- General misconception
+
+  Gateways are usually running outside the cluster as a first entry point. The cluster should get rid of the data as fast as possible as persistence cannot be provided here (things like Kafka would again run outside). Try to compensate for short-lived hiccups. Otherwise, point back the backpressure to the client and do not try to solve the big picture.
+
+## Proposal
+
+Most of the drawbacks can be solved by running the gateway logic node-local only. Every instance always processes only data of the local node. With that a natural scalability is given and can be extended by vertical scaling capabilities. The "old" gateways will be running in agent mode but still called "OTLP Gateway". Additionally, there will be a new uniform service `telemetry-otlp` forwarding to the node-local entity using the [internalTrafficPolicy: local](https://kubernetes.io/docs/reference/networking/virtual-ips/#internal-traffic-policy) setting. Existing services can stay compatible using the same approach.
+
+![arch](./../assets/otlp-gateway-new.drawio.svg)
+
+### Remaining Drawbacks of the Agent Approach  
+
+#### Coupling of Signal Types  
+
+Decoupling of signal types will no longer be possible, which may result in scenarios where logs are dropped due to metric overload. However, with proper hardening, this should not pose a significant issue.  
+
+Let's compare a **shared OTLP agent setup** with an **agent-per-signal setup** in terms of robustness. The OTel Collector offers two key mechanisms to prevent overload:
+
+1. **Memory Limiter**  
+   - Applied globally, no way to restrict it to a specific pipeline or signal type.
+   - Performs periodic checks of memory usage and will begin refusing data and forcing GC to reduce memory consumption when defined limits have been exceeded. It influenced by:
+     - Heap allocated object held by the Collector.  It can be the OTLP exporter sending queues or various caches used by different components in the chain (e.g. Kubernetes Attributes Processor informers cache).  
```

Comment:
Not sure what the following terms mean, we didn't introduce them earlier:

- **GC** - is this a collector? Gateway Collector?
- **Heap allocated object** - is this an object that's heap-allocated? Then I believe we need a hyphen because "heap allocated" is used as an adjective.
- **Kubernetes Attributes Processor informers cache** - is this the cache of the/an informer of the Kubernetes Attributes Processor? Then maybe it should be "Kubernetes Attributes Processor informer's cache", but even clearer: "Cache of the Kubernetes Attributes Processor informer". Or is it the informer cache of the Kubernetes Attributes Processor? Then drop the "s": "Kubernetes Attributes Processor informer cache"